pages/protocol-config.mdx

```diff
@@ -0,0 +1,57 @@
+
+- Predictable fees: Setting a limit prevents users from accidentally creating transactions with exorbitant costs due to unbounded execution steps, facilitating predictable and manageable transaction fees.
+- Fairness and stability: Limiting the step count promotes fair resource sharing among users, preventing any single transaction from consuming excessive resources and impacting network stability.
+- Adherence to block times: The maximum step limit also helps ensure that transactions fit within the designated block time, contributing to the network's overall predictability and stability.
+
+In essence, `InvokeTxMaxNSteps` plays a critical role in allowing app chain developers to configure their Dapp for security and performance.
+
+### Configuring ValidateMaxNSteps
+Starknet's `__validate__` function acts as a pre-execution check. It receives a user's calls and executes validation to verifies adherence to the contract's constraints. 
+It returns "VALID" to signify successful validation, else aborting the transaction, preventing unauthorized or invalid operations. 
+
+`ValidateMaxNSteps` enforces a maximum step limit within the `__validate__` function safeguarding against DoS attacks.  
+Since failed `__validate__` calls don't incur fees, malicious actors could exploit them to overload the network with invalid transactions. 
+While KYC and other measures can be employed, limiting `__validate__` logic complexity remains crucial in mitigating DoS threats.
+
+### Configuring MaxRecursionDepth
+MaxRecursionDepth serves as a critical safeguard against infinite loops within Cairo contracts. It defines the maximum number of times a function can call itself recursively before execution is halted. 
+
+This parameter is essential for:
+
+- Preventing DoS Attacks: Malicious actors could exploit uncapped recursion to create infinite loops, consuming excessive resources and potentially halting network operations.
+- Encouraging Efficient Code: By imposing a limit on recursion, developers are incentivized to write code that utilizes iterative approaches or alternative control flow mechanisms.
+
+Overall, MaxRecursionDepth plays a crucial role in maintaining the security, stability, and efficiency of the Starknet network by preventing infinite loops and encouraging sound programming practices within contracts.
+
+### Configuring ProgramHash
+The ProgramHash field is a cryptographic identifier of the specific Starknet OS version used to generate proofs for L1 verification. These proofs are submitted to the L1 Ethereum layer for verification.
+
+This field must match the latest hash supported by Madara allowing for consistency in proof generation ensuring that when the Starknet OS undergoes an upgrade, the app chain is updated accordingly. 
+This ensures all newly generated proofs utilize the latest version, enabling the network to evolve securely while maintaining compatibility with existing infrastructure.
+
+### Configuring DisableTransactionFee
```

Comment:
we should add a warning that this does open a vector fro DoS and spam and app chains should be careful when dealing with this